task/mgmt-gateway/src/mgs_gimlet.rs

```diff
@@ -80,15 +95,18 @@ impl MgsHandler {
         // avoid blocking while the entire erase happens. If we're in that case,
         // set our timer for 1 tick from now to give a window for other
         // interrupts/notifications to arrive.
-        if self.host_flash_update.needs_sectors_erased() {
+        if self.host_flash_update.is_preparing()
+            || self.sp_update.is_preparing()
+        {
             Some(sys_get_timer().now + 1)
         } else {
             self.usart.from_rx_flush_deadline
         }
     }
 
     pub(crate) fn handle_timer_fired(&mut self) {
-        self.host_flash_update.erase_sectors_if_needed();
+        self.host_flash_update.step_preparation();
```

Comment:
Add a comment here that these are no-ops if no update is in progress, and that only one will be running at a time.

Reply:
Added in 644a64a3